docs/src/pages/ShorthandProps.mdx

```diff
@@ -0,0 +1,186 @@
+
+Thus, in its simplest form, it could be used the following way:
+
+```jsx
+<Button icon={() => <Icon name="chess rook" />} />
+```
+
+However, if it will be just about that, this function form wouldn't introduce any additional value. So, lets take a look on the scenarios where it is really helpful.
+
+### 'Render' callback argument
+
+When function is provided as a shorthand value, there is a `render` callback provided
+as its argument. This `render` argument is a function that 'knows' how to render all
+the previously considered shorthand types - i.e. primitive values, objects, React Elements.
+
+Here is an example that represents this:
+
+```jsx
+<>
+  {/* All three have the same effect: */}
+  <Button icon="chess rook" />
+  {/* render() transforms string to <Icon /> element */}
+  <Button icon={render => render('chess rook')} />
+  {/* render() transforms object to <Icon /> element */}
+  <Button icon={render => render({ name: 'chess rook' })} />
+</>
+```
+
+### Handle Async data loading scenarios
+
+This 'render callback' arg makes it possible for the client to tackle scenarios where slot's data may be fetched asynchronously.
+
+Consider the following scenario: suppose we need to render `Button`'s icon, but the
+problem is that icon's name is fetched from some remote source. Thus, we cannot do it as
+simply as that, as icon's name is not known at the moment `Button` is rendered:
+
+```jsx
+<Button icon={{ name: /* unknown yet */ undefined }} />
+```
+
+It is quite common case that there is some component that is responsible for data fetching
+abstraction - and it is possible to 'tell' this component what should be rendered while data
+is loading, as well as what should be rendered when data is fetched.
+
+```jsx
+<AsyncData
+  getData={fetch('https://some/url')}
+  onLoading={<div>Loading..</div>} // renders when data is loading
+  onLoaded={data => <div>Loaded Data: {data}</div>} // renders once data is fetched
+/>
+```
+
+Lets suppose that we want to render `loading` icon before data is fetched, and switch it with
+the icon which `name` will be provided by fetched `data` object. Also, note
+that we have `render` arg that knows _how_ to render `Button`'s icons.
+
+Putting it all together, we will address our async data fetch case:
+
+```jsx
+<Button
+  icon={render => (
+    <AsyncData
+      getData={fetch('https://some/url')}
+      onLoading={<Icon name="loading" />}
+      /* using 'render' arg here to render icon from loaded na */
+      onLoaded={data => render({ name: data.iconName })}
+    />
+  )}
+/>
+```
+
+Note that this example can be further simplified, as `render` can be used to render
+'loading' icon as well:
+
+```jsx
+<Button
+  icon={render => (
+    <AsyncData
+      getData={fetch('https://some/url')}
+      /* using 'render' arg here to render loading icon as well */
+      onLoading={render('loading')}
+      onLoaded={data => render({ name: data.iconName })}
+    />
+  )}
+/>
+```
+
+### Customizing rendered tree
```

Comment:
We should also add new use case now - completely replace slot content.

Reply:
Actually, the example in this section does required thing. I removed a paragraph that doesn't have any sense now, https://github.com/stardust-ui/react/commit/1fbc89a1f148398e7aa2836603a8b540dd48ffb4